client/python/lib/test/test_tensors.py

```diff
@@ -113,6 +113,14 @@ def test_make_tensor_proto_valid_int():
     assert tensor_proto.dtype == DataType.DT_INT8
     assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
 
+def test_make_tensor_proto_valid_int_reshape():
+    values = [1, 2, 3]
+    tensor_proto = make_tensor_proto(shape=[4], dtype = DataType.DT_INT8, values = values)
+    np_values = np.array(values, np.int8)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_INT8
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=4)])
```

Comment:
is it possible to reshape 3 elements to [4]?

Reply:
Yes, it is possible.
With values = [1,2,3] and shape = [4] the tensor_proto from output looks like this:
```
dtype: DT_INT32
tensor_shape {
  dim {
    size: 4
  }
}
tensor_content: "\001\000\000\000\002\000\000\000\003\000\000\000"
```

Reply:
`tensor_shape` is not connected to any other field either `tensor_content` or `val` containers.
Therefore we need to make sure the shape value and the actual data are adjusted throughout our processing.